### ISSUE TITLE
Figure out how to runtime output with the test shared framework

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#15936]
Today we have a runtime output "bin\runtime\netcoreapp-Windows_NT-Debug-x64" and we have a runtime folder "testhost\netcoreapp-Windows_NT-Debug-x64\shared\Microsoft.NETCore.App\9.9.9" output so that we can run our tests as a shared application. The contents of those 2 directories are currently duplicated. We should figure out a way to hard-line the entire directory (not just the individual files) or a way to configure dotnet.exe such that we can tell it to use our runtime output path as the shared framework.

cc @karajas 

@gkhanna79 do you know if there is any way to configure dotnet.exe to use a particular folder as the shared framework? If not do you think that is a feature request that might be doable?

**Comment by gkhanna79** [comment]
Currently, there is not a way to specify that since the host has bunch of invariants built around it being next to shared folder.

Is there a reason you cannot use the dotnet.exe from the second location above?

**Comment by weshaggard** [comment]
>Is there a reason you cannot use the dotnet.exe from the second location above?

If we do that how can we specify the additional test directory. We essentially have the shared framework and test output directory we need to be able to resolve from both.

**Comment by gkhanna79** [comment]
Are the additional test artifacts to be consumed part of the app.deps.json?

**Comment by weshaggard** [comment]
No the aren't part of the app.deps.json file. The deps file we have today only contains bits from the shared runtime folder. We don't generate a unique deps file for each test project, we could potentially consider doing that if necessary but I would prefer it if we don't need to.

**Comment by ViktorHofer** [comment]
related to enabling VS Test Explorer where we could also specify a custom shared runtime folder instead of directing to our dotnet.exe in corefx. But I think there is a lot of value in fixing the original issue itself as the current duplication of the testhost folder is unfortunate.

@danmosemsft who is currently the right person to contact for such a feature addition to dotnet.exe?

**Comment by danmosemsft** [comment]
@ViktorHofer it comes from the core-setup repo, so probably would be an issue there. @steveharter would be a good person to talk to about it.

**Comment by ericstj** [comment]
Yeah, I just hit this yesterday when trying to remove buildtools and ran into the deps.json generation task which is another ugly symptom of this problem.   Let's meet with should set up a meeting with @jeffschwMSFT and @vitek-karas to discuss.

Do you think you could write up a) our key requirements b) problems with current approach?

**Comment by ViktorHofer** [comment]
Sure, will put that on my agenda for today.

**Comment by wtgodbe** [comment]
@ericstj @ViktorHofer was there any update here?

**Comment by ViktorHofer** [comment]
For getting rid of the dotnet in the testhost folder I opened https://github.com/dotnet/core-setup/issues/4915. Getting rid of the runtime directory is to-do and is tracked by this issue. @wtgodbe do you want to pick that issue? The goal is to get rid of the runtime directory (disabled the binplacing into it) and just have the testhost directory.

**Comment by wtgodbe** [comment]
Sure thing, I'll look into it when I get a chance.

**Comment by ViktorHofer** [comment]
@wtgodbe I took care of this in https://github.com/dotnet/corefx/pull/38792/commits/b1619b162b0b3116c59883368bdae87dbe31b5be, I hope you don't mind.

**Comment by ViktorHofer** [comment]
A possible solution for this is tracked in https://github.com/dotnet/core-setup/issues/4809
